tests/callbacks_test.go

```diff
@@ -238,3 +241,122 @@ func (s *FunctionalSuite) TestWorkflowNexusCallbacks_CarriedOverContinueAsNew()
 		}
 	}
 }
+
+func (s *FunctionalSuite) TestWorkflowCallbacks_ForceActivityComplete() {
+	dc := s.testCluster.host.dcClient
+	dc.OverrideValue(s.T(), dynamicconfig.FrontendEnableCallbackAttachment, true)
+	defer dc.RemoveOverride(dynamicconfig.FrontendEnableCallbackAttachment)
+
+	sdkClient, err := client.Dial(client.Options{
+		HostPort:  s.testCluster.GetHost().FrontendGRPCAddress(),
+		Namespace: s.namespace,
+	})
+	s.NoError(err)
+
+	pp := temporalite.NewPortProvider()
+	callbackAddress := fmt.Sprintf("localhost:%d", pp.MustGetFreePort())
+	s.NoError(pp.Close())
+
+	ach := &serveHTTPHandler{
+		sdkClient:         sdkClient,
+		requestCompleteCh: make(chan error, 1),
+		activityErr:       make(chan error, 1),
+	}
+
+	mockErrorActivity := func(ctx context.Context) error {
+		ai := activity.GetInfo(ctx)
+		formData := url.Values{}
+		formData.Add("name_space", ai.WorkflowNamespace)
+		formData.Add("workflow_id", ai.WorkflowExecution.ID)
+		formData.Add("run_id", ai.WorkflowExecution.RunID)
+		formData.Add("activity_id", ai.ActivityID)
+
+		httpUrl := fmt.Sprintf("http://%s", callbackAddress)
+		_, err := http.PostForm(httpUrl, formData)
+		if err != nil {
+			panic(err)
+		}
+		err = errors.New("mock error")
+		ach.activityErr <- err
+		return err
```

Comment:
Just return an error here.

Reply:
So the reason I pass those information to the HttpServer is for that server to make a call with `CompleteActivityByID`, another way we can use here is to make up those inforamiont and cache it as a global variable for the HttpServer to access, any thoughts on that?

Reply:
You're not testing the right thing here, the activity attempt is still in progress. You want to check that the activity can complete when it is backing off.
And the http server isn't adding anything, you can call the client from any goroutine.